packages/core/strapi/src/Strapi.ts

```diff
@@ -505,87 +623,79 @@ class Strapi {
 
     this.isLoaded = true;
 
-    return this;
+    return this as this & Required<StrapiI>;
   }
 
   async startWebhooks() {
-    const webhooks = await this.webhookStore.findWebhooks();
-    webhooks.forEach((webhook) => this.webhookRunner.add(webhook));
-  }
-
-  reload() {
-    const state = {
-      shouldReload: 0,
-    };
-
-    const reload = function () {
-      if (state.shouldReload > 0) {
-        // Reset the reloading state
-        state.shouldReload -= 1;
-        reload.isReloading = false;
-        return;
-      }
-
-      if (this.config.get('autoReload')) {
-        process.send('reload');
-      }
-    };
-
-    Object.defineProperty(reload, 'isWatching', {
-      configurable: true,
-      enumerable: true,
-      set(value) {
-        // Special state when the reloader is disabled temporarly (see GraphQL plugin example).
-        if (state.isWatching === false && value === true) {
-          state.shouldReload += 1;
-        }
-        state.isWatching = value;
-      },
-      get() {
-        return state.isWatching;
-      },
-    });
-
-    reload.isReloading = false;
-    reload.isWatching = true;
+    const webhooks = await this.webhookStore?.findWebhooks();
+    if (!webhooks) {
+      return;
+    }
 
-    return reload;
+    for (const webhook of webhooks) {
+      this.webhookRunner?.add(webhook);
+    }
   }
 
-  async runLifecyclesFunctions(lifecycleName) {
+  async runLifecyclesFunctions(lifecycleName: 'register' | 'bootstrap' | 'destroy') {
     // plugins
     await this.container.get('modules')[lifecycleName]();
 
     // admin
-    const adminLifecycleFunction = this.admin && this.admin[lifecycleName];
+    const adminLifecycleFunction = this.admin?.[lifecycleName];
     if (isFunction(adminLifecycleFunction)) {
       await adminLifecycleFunction({ strapi: this });
     }
 
     // user
-    const userLifecycleFunction = this.app && this.app[lifecycleName];
+    const userLifecycleFunction = this.app[lifecycleName];
     if (isFunction(userLifecycleFunction)) {
       await userLifecycleFunction({ strapi: this });
     }
   }
 
-  getModel(uid) {
-    return this.contentTypes[uid] || this.components[uid];
+  getModel(uid: Common.UID.ContentType): Schema.ContentType;
+  getModel(uid: Common.UID.Component): Schema.Component;
+  getModel<TUID extends Common.UID.Schema>(uid: TUID): Schema.ContentType | Schema.Component {
+    if (uid in this.contentTypes) {
+      return this.contentTypes[uid as Common.UID.ContentType];
+    }
+
+    if (uid in this.components) {
+      return this.components[uid as Common.UID.Component];
+    }
+
+    throw new Error('Model not found');
```

Comment:
Unfortunately some code is relying on this returning `undefined`, for example https://github.com/strapi/strapi/blob/a2a179e35db31b9ef7383d89630b131422457c19/packages/plugins/graphql/server/services/builders/filters/content-type.js#L26 which causes strapi to be unable to start because it's looking for a model without checking if it's a morph relation first.